### PR TITLE
netcat tests

### PR DESCRIPTION
**Pull request by binjip978**

### CODE REVIEW

File: cmds/core/netcat/netcat.go [hunk 1]
```diff
@@ -16,59 +16,106 @@ import (
...
+type cmd struct {
+	stdin  io.Reader
+	stdout io.Writer
+	stderr io.Writer
+	addr   string
+	params
+}
+
+func command(stdin io.Reader, stdout io.Writer, stderr io.Writer, args []string, p params) (*cmd, error) {
+	if len(args) != 2 {
```
Comment (rminnich) [inline cmds/core/netcat/netcat.go:53]: what if there are 3 args? It's not a missing name in that case I guess? It's too many?

File: cmds/core/netcat/netcat.go [hunk 2]
```diff
@@ -16,59 +16,106 @@ import (
...
+	port    string
+}
+
+func parseParams() params {
+	netType := flag.String("net", "tcp", "What net type to use, e.g. tcp, unix, etc.")
+	listen := flag.Bool("l", false, "Listen for connections.")
+	verbose := flag.Bool("v", false, "Verbose output.")
+	flag.Parse()
+
+	return params{
```
Comment (rminnich) [inline cmds/core/netcat/netcat.go:37]: I am wondering if params should not just include the host and port too?

**Review by rminnich — COMMENTED** [review]
This is a very nice improvement, I am just wondering about the params struct and args.
Also if people type
netcat host port blalalb
and get
"missing hostname or port"
that will be very confusing. I guess I still prefer the go style address, 

should flag.Arg()s be handled in parseParams?

File: cmds/core/netcat/netcat.go (hunk 1 above)
Comment (binjip978) [reply cmds/core/netcat/netcat.go:53]: Yeah, checking for != 2 was a mistake, interstingly that on linux `nc -l localhost 8080 xxx` will return error and print usage, on macOS it just ignore it. But if you prefer Go style I will revert it to previous style.

File: cmds/core/netcat/netcat.go (hunk 2 above)
Comment (binjip978) [reply cmds/core/netcat/netcat.go:35]: They can, but in find it easier to do error handling when parseParams works only with flag arguments and deal with non flag arguments later. But I can change that if needed.

**Pull request merged**